internal/parser/planparserv2/utils.go

```diff
@@ -307,7 +320,7 @@ func handleCompare(op planpb.OpType, left *ExprWithType, right *ExprWithType) (*
 }
 
 func relationalCompatible(t1, t2 schemapb.DataType) bool {
-	both := typeutil.IsStringType(t1) && typeutil.IsStringType(t2)
+	both := (typeutil.IsStringType(t1) || typeutil.IsJSONType(t1)) && (typeutil.IsStringType(t2) || typeutil.IsJSONType(t2))
```

Comment:
questions? what is the relationalCompatible mean?
Json can be compared with string or not?
Json with int?
String with int?